src/bsoncxx/test/json.cpp

```diff
@@ -93,24 +94,58 @@ TEST_CASE("CXX-1246: Legacy Extended JSON (Explicit)") {
 
 TEST_CASE("CXX-1246: Relaxed Extended JSON") {
     using namespace bsoncxx;
+    // TODO CXX-2227: Remove skip after minimum libmongoc version is bumped.
+    if (!mongoc_version) {
+        WARN("Skipping — environment variable MONGOC_VERSION must be set");
+        return;
+    }
+
     types::b_binary bin_val{
         binary_sub_type::k_uuid, 8, reinterpret_cast<const uint8_t*>("deadbeef")};
     auto doc = make_document(kvp("number", 42), kvp("bin", bin_val));
     auto output = to_json(doc.view(), ExtendedJsonMode::k_relaxed);
-    REQUIRE(
-        output ==
-        R"({ "number" : 42, "bin" : { "$binary" : { "base64": "ZGVhZGJlZWY=", "subType" : "04" } } })");
+
+    // TODO CXX-2227: Remove conditional result after minimum libmongoc version is bumped.
+    //
+    // As of libmongoc 1.18.0, "base64" has correct spacing (see CDRIVER-3958) after extJSON
+    // marshalling.
+    const char* expected;
+    if (std::string(mongoc_version).compare("1.18.0") >= 0) {
```

Comment:
Can you change the comparison to use the integer macros exposed in `mongoc-version.h`?

E.g. `if ( (MONGOC_MAJOR_VERSION == 1 && MONGOC_MINOR_VERSION >= 18) || MONGOC_MAJOR_VERSION > 1)`

`mongoc-version.h` should be included through `mongoc.h` already.

A string comparison won't do a lexicographic compare.

Reply:
Discussed offline. We actually want to `BSON_VERSION` macros from `bson-version.h` since this is in `bsoncxx`. Used those instead of the string comparison and environment variable (changes to `.mci.yml` removed).